soc: st: stm32: adding option to enable prefetch buffer

lkoudelka opened the pull request:
For more effective code execution on STM32 devices is convenient to enable flash prefetch buffer. To be enabled by default, possible to disable using kconfig.

ajarmouni-st commented on soc/st/stm32/Kconfig:
```diff
@@ -46,6 +46,9 @@ config SWJ_ANALOG_PRIORITY
 	  This priority must be greater than GPIO_INIT_PRIORITY and lower than
 	  UART_INIT_PRIORITY.
+	  
+config FLASH_PREFETCH
```
```suggestion
config STM32_FLASH_PREFETCH
```

ajarmouni-st commented on soc/st/stm32/stm32wbax/soc.c:
```diff
@@ -38,6 +38,9 @@ int stm32wba_init(void)
 	LL_ICACHE_SetMode(LL_ICACHE_1WAY);
 	LL_ICACHE_Enable();
 
+#if defined(CONFIG_FLASH_PREFETCH)
```
```suggestion
#ifdef CONFIG_FLASH_PREFETCH
```

ajarmouni-st commented on soc/st/stm32/Kconfig:
```diff
@@ -46,6 +46,9 @@ config SWJ_ANALOG_PRIORITY
 	  Initialization priority of the routine within the PRE_KERNEL1 level.
 	  This priority must be greater than GPIO_INIT_PRIORITY and lower than
 	  UART_INIT_PRIORITY.
+	  
```
trailing whitespace

erwango commented on soc/st/stm32/stm32wbax/soc.c:
```diff
@@ -38,6 +38,9 @@ int stm32wba_init(void)
 	LL_ICACHE_Enable();
 
+#if defined(CONFIG_FLASH_PREFETCH)
+	__HAL_FLASH_PREFETCH_BUFFER_ENABLE();
```
Please use LL

lkoudelka replied:
Only HAL macro definition is for now available in the libraries, no LL equivalent right now

erwango replied:
ok, let's go with this version for now, but an issue should be raised on the LL then.

erwango reviewed (dismissed):
Great! Please extend to other compatible series (L4, ...)

erwango commented:
Dismissing my request to extend on other series.
But please fix compliance guidelines checks and align on @ajarmouni-st's requests

lkoudelka committed:
soc: st: stm32: adding option to enable prefetch buffer

For more effective code execution on STM32 devices is convenient
 to enable flash prefetch buffer.
To be enabled by default, possible to disable using kconfig.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>
